src/vs/workbench/contrib/search/browser/quickTextSearch/textSearchQuickAccess.ts

```diff
@@ -128,7 +128,8 @@ export class TextSearchQuickAccess extends PickerQuickAccessProvider<IPickerQuic
 
 		const getAsyncResults = async () => {
 			await result.asyncResults;
-			return this.searchModel.searchResult.matches().filter(e => result.syncResults.indexOf(e) === -1);
+			const syncResultURIs = result.syncResults.map(e => e.resource);
+			return this.searchModel.searchResult.matches().filter(e => syncResultURIs.indexOf(e.resource) === -1);
```

Comment:
This looks quite expensive of an action, could sync results also be stored in a map/set such that looking up a match is O(1)?